### PR TITLE
Adds editor config

### PR DESCRIPTION
**Pull request by jesse-peters**
This PR adds [editorconfig](https://editorconfig.org/) - a tool to help enforce formatting.

Most IDEs either have native support, or have a plugin available. The pipeline will enforce the formatting before a merge is allowed.

### CODE REVIEW

File: cmd/version/version_test.go [hunk 1]
```diff
@@ -47,17 +47,17 @@ func TestVersionCMD(t *testing.T) {
 			Args:            []string{},
 			IsErrorExpected: false,
 			ExpectedMessage: `{
-    "buildTime": "2023-01-07",
-    "version": "v0.1.0",
-    "commit": "a7c46aa017bc447ece506629196bd0548cbbc469"
+	"buildTime": "2023-01-07",
+	"version": "v0.1.0",
+	"commit": "a7c46aa017bc447ece506629196bd0548cbbc469"
```
Comment (RRethy) [inline cmd/version/version_test.go:52]: We can change this to string concatenation so tests don't fail.

```go
`    "buildTime": "2023-01-07",` +
...
```

File: pkg/cli/printer/json.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ func (printer *JSON) Format() string {
 }
 
 func (printer *JSON) Print(w io.Writer, obj any) error {
-	data, err := json.MarshalIndent(obj, "", "    ")
+	data, err := json.MarshalIndent(obj, "", "	")
```
Comment (RRethy) [inline pkg/cli/printer/json.go:22]: Json should be indented with spaces.

Comment (jesse-peters) [reply pkg/cli/printer/json.go:22]: Alright thanks, I was just about to look it up after I saw your first comment. I'll get that fixed.

**Review by RRethy — APPROVED** [review]
Outside of the JSON comment, LGTM

**Review by krisnova — APPROVED** [review]

**Pull request merged**